src/account/SupportContact.tsx

```diff
@@ -77,28 +91,40 @@ function SupportContact({ route }: Props) {
     if (attachLogs) {
       attachments = await Logger.getLogsToAttach()
       if (attachments) {
-        email.attachments = attachments
-        email.body += (email.body ? '<br/><br/>' : '') + '<b>Support logs are attached...</b>'
+        emailInfo.attachments = attachments
+        emailInfo.body +=
+          (emailInfo.body ? '<br/><br/>' : '') + '<b>Support logs are attached...</b>'
       }
     }
-    // Used to prevent flickering of the activity indicator on quick uploads
-    setTimeout(() => setInProgress(false), 1000)
     try {
-      await sendEmail(
-        email,
-        deviceInfo,
-        // Get the current months log file to attach as text if sendEmailWithNonNativeApp is used
-        attachments
-          ? attachments.find(
-              (attachment: { name: string }) => attachment.name === Logger.getCurrentLogFileName()
-            )?.path ?? false
-          : false
-      )
+      if (useZendeskApi) {
+        await sendSupportRequest({
+          message,
+          deviceInfo,
+          logFiles: attachments ?? [],
+          userEmail: email,
+          userName: name,
+          subject: t('supportEmailSubject', { appName: APP_NAME, user: userId }),
+        })
+      } else {
+        await sendEmail(
+          emailInfo,
+          deviceInfo,
+          // Get the current months log file to attach as text if sendEmailWithNonNativeApp is used
+          attachments
+            ? attachments.find(
+                (attachment: { name: string }) => attachment.name === Logger.getCurrentLogFileName()
+              )?.path ?? false
+            : false
+        )
+      }
+      // Used to prevent flickering of the activity indicator on quick uploads
+      setTimeout(() => setInProgress(false), 1000)
```

Comment:
this isn't really related to the new changes, but it seems like a Promise.all between this and the email would make sense here, so if the email takes more than a second by itself, this doesn't add an unnecessary delay

Reply:
The timeout also has a second use b/c navigateBackAndToast is a bit slow and without the timeout the page briefly flashes the done-loading state before redirecting. I added a comment explaining.